test/Microsoft.ML.CpuMath.PerformanceTests/AvxPerformanceTests.cs

```diff
@@ -11,6 +11,26 @@ namespace Microsoft.ML.CpuMath.PerformanceTests
 {
     public class AvxPerformanceTests : PerformanceTests
     {
+        private AlignedArray _testMatrices;
+        private AlignedArray _testSrcVectors;
+        private AlignedArray _testDstVectors;
+
+        [GlobalSetup(Targets = new string[] { nameof(MatMulX), nameof(MatMulTranX) })]
+        public void MatMulSetup()
+        {
+            Setup();
+            int vectorAlignment = CpuMathUtils.GetVectorAlignment();
+
+            _testMatrices = new AlignedArray(1000 * 1000, vectorAlignment);
```

Comment:
Using `AlignedArray` is likely going to remove most of the overhead that the unaligned path would have given.

Reply:
even when we are not using the aligned array functions like compat and getBaseaddress or moveData ?

Reply:
I'm not sure with the inner details of how `AlignedArray` works, but it would probably just be better to not use it at all.

Reply:
But if we dont use aligned array , how we will be able to test the before case as it expects the data to be aligned

Reply:
(especially since `MatMulX` which takes an `AlignedArray` does use `getBaseAddress` etc)

Reply:
The before case likely still needs to use `AlignedArray`, but the after case is supposed to remove the need for `AlignedArray` entirely and using `AlignedArray` will add some amount of overhead for re-aligning the data which will skew the after numbers from the "real world" case.